udata/migrations/2017-10-24-cleanup-resources-checksum.js

```diff
@@ -0,0 +1,22 @@
+/*
+ * Remove checksum from resources where it has no value
+ */
+
+var nbCleaned = 0;
+
+db.dataset.find({
+    $and: [
+        {'resources.checksum': {$exists: true}},
+        {'resources.checksum.value': {$exists: false}},
+    ]
+}).forEach(dataset => {
+    dataset.resources.forEach(resource => {
+        if (resource.checksum && !resource.checksum.value) {
+            delete resource.checksum;
+            nbCleaned++;
+        }
+        db.dataset.save(dataset);
```

Comment:
You save the dataset for each resource, do it only once by moving the `db.dataset.save()` outside the `dataset.resources.forEach()`